build.gradle

```diff
@@ -69,9 +69,14 @@ dependencies {
 
     dependencyManagement {
         dependencies {
+            dependency group: 'org.springframework', name: 'spring-aop', version: springFrameworkVersion
             dependency group: 'org.springframework', name: 'spring-beans', version: springFrameworkVersion
+            dependency group: 'org.springframework', name: 'spring-context', version: springFrameworkVersion
+            dependency group: 'org.springframework', name: 'spring-core', version: springFrameworkVersion
             dependency group: 'org.springframework', name: 'spring-expression', version: springFrameworkVersion
-            dependency group: 'org.springframework', name: 'spring-aop', version: springFrameworkVersion
+            dependency group: 'org.springframework', name: 'spring-jcl', version: springFrameworkVersion
```

Comment:
why do we need to add all of these explicit dependencies when they were not previously added?  if these are oerriding transitive versions, there is a better way to do that, rather then add explicit 1st level dependencies (which can cause other problems) - see https://docs.gradle.org/7.6.2/userguide/dependency_constraints.html#sec:adding-constraints-transitive-deps

Reply:
The dependencyManagement does not introduce direct dependencies (if that's what you mean by 1st level dependencies).
To override versions of transitive dependencies, we can not use gradle constrains _because_ the project uses spring [Dependency Management Plugin](https://docs.spring.io/dependency-management-plugin/docs/1.1.0/reference/html). It overrides any constraints we define. (I've verified this by adding constraints instead of dependencyManagement)
Ref - https://github.com/spring-projects/spring-boot/issues/26840#issuecomment-857865885
We had added this block in PR #3 & added more to this in this PR.

One more way to override the versions managed spring boot is by setting [version properties](https://docs.spring.io/spring-boot/appendix/dependency-versions/properties.html#appendix.dependency-versions.properties).

We've used some these properties to override versions for tomcat, jackson etc. There's a version property for spring-framework as well. Ideally we should be using this property to override versions for all of these spring-* libraries.

If we use this property, spring boot first needs to fetch [spring-framework-bom](https://github.com/spring-projects/spring-framework/tree/main/framework-bom) & then use versions declared in the bom.

WSO has published version 5.3.36-wso2v1 the spring-* libraries that we've used  but not for the bom itself. Due to this, we need to declare each dependency separately in the `dependencyManagement` block.
Ref:
 - https://maven.wso2.org/nexus/content/repositories/public/org/springframework/
 - https://maven.wso2.org/nexus/content/repositories/releases/org/springframework/